UI: Prevent recursive aliases from being resolved. Fixes bitburner-official#630

ficocelliguy opened the pull request:
Refactor alias processing: when resolving an alias, look at the tracked list of previously-processed aliases that led to the current one, and do not resolve any ~~aliaii~~ aliases already present in it. This ensures there are no loops in the alias reference tree in a given command, and replaces the current only-one-alias-lookback solution.

closes bitburner-official#630

ficocelliguy committed:
Prevent recursive aliases from being resolved. Fixes bitburner-official#630

ficocelliguy committed:
Updated logic to track command replaced instead of replaced alias; improved tests

Snarling commented:
Why the max depth check? I think the origin of that was just to prevent recursive replacements, but if you're already preventing recursive replacements then aliases should just continue to replace until there are no more non-recursive replacements to make. e.g. if a=b, b=c, c=d [...] y=z, then `a` should be replaced with `z`, it shouldn't randomly stop replacing aliases 10 levels deep.

ficocelliguy committed:
remove max depth from alias resolution logic

ficocelliguy commented:
I had left in the max depth as a bit of a failsafe (and in case it was an existing feature). I have removed it and updated the unit tests accordingly.

ficocelliguy committed:
Fix linting
